language-support/hs/bindings/src/DA/Ledger/LedgerService.hs

```diff
@@ -23,3 +25,13 @@ runLedgerService (LedgerService r) ts cc = runReaderT r (ts,cc)
 
 makeLedgerService :: (TimeoutSeconds -> ClientConfig -> IO a) -> LedgerService a
 makeLedgerService f = LedgerService $ ReaderT $ \(ts,cc) -> ledgerRetry $ f ts cc
+
+withTimeout :: LedgerService a -> LedgerService a
+withTimeout service = do
+    (ts,cc) <- LedgerService ask
+    Just x <- liftIO $ timeout (fromIntegral ts) (runLedgerService service ts cc)
+    return x
+
+mapIOLedgerService :: (IO a -> IO b) -> LedgerService a -> LedgerService b
```

Comment:
I think a better solution here would be to remove this function, make `LedgerService` an instance of `MonadUnliftIO` and then use the generalized exception handling functions in `unliftio`, e.g. [bracket](https://hackage.haskell.org/package/unliftio-0.2.11/docs/UnliftIO-Exception.html#v:bracket).

Reply:
I realised I want `bracket` but not in the `IO` monad. 
I discovered `MonadUnliftIO` last night, and it was on my my list to learn about today...

Reply:
`LedgerService` now an instance of MonadUnliftIO.
`mapIOLedgerService` is gone